### PR TITLE
Added kahlan support

### PR DESCRIPTION
**Pull request by Grafikart**
| Q             | A
| ------------- | ---
| Branch        | master
| Bug fix?      | no
| New feature?  | yes
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | yes
| Documented?   | yes
| Fixed tickets | #225 

Added support for Kahlan.

# New Task Checklist:

- [x] Is the README.md file updated?
- [x] Are the dependencies added to the composer.json suggestions?
- [x] Is the doc/tasks.md file updated?
- [x] Are the task parameters documented?
- [x] Is the task registered in the tasks.yml file?
- [x] Does the task contains phpspec tests?
- [x] Is the configuration having logical allowed types?
- [x] Does the task run in the correct context?
- [x] Is the `run()` method readable?
- [x] Is the `run()` method using the configuration correctly?
- [x] Are all CI services returning green?

### CODE REVIEW

File: doc/tasks/kahlan.md [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+          src: ~
+          spec: ~
+          pattern: ~
+          reporter: ~
+          coverage: ~
+          clover: ~
+          istanbul: ~
+          lcov: ~
+          ff: ~
+          'no-colors': ~
```
Comment (Grafikart) [inline doc/tasks/kahlan.md]: I'm not sure on this syntax still

Comment (veewee) [inline doc/tasks/kahlan.md]: In other tasks we use the underscore instead so that we do not need quotes.

File: src/GrumPHP/Task/Kahlan.php [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+        $files = $context->getFiles()->name('*.php');
+        if (0 === count($files)) {
+            return TaskResult::createSkipped($this, $context);
+        }
+
+        $arguments = $this->processBuilder->createArgumentsForCommand('kahlan');
+        $arguments->add('--no-interaction');
+
+        $process = $this->processBuilder->buildProcess($arguments);
+        $process->run();
```
Comment (veewee) [inline src/GrumPHP/Task/Kahlan.php]: You are not using the configured values at the moment.

File: src/GrumPHP/Task/Kahlan.php [hunk 3]
```diff
@@ -0,0 +1,102 @@
...
+        $resolver->addAllowedTypes('src', ['null', 'string']);
+        $resolver->addAllowedTypes('spec', ['null', 'string']);
+        $resolver->addAllowedTypes('pattern', ['null', 'string']);
+        $resolver->addAllowedTypes('reporter', ['null', 'string']);
+        $resolver->addAllowedTypes('coverage', ['null', 'string', 'int']);
+        $resolver->addAllowedTypes('clover', ['null', 'string']);
+        $resolver->addAllowedTypes('istanbul', ['null', 'string']);
+        $resolver->addAllowedTypes('lcov', ['null', 'string']);
+        $resolver->addAllowedTypes('ff', ['null', 'int']);
+        $resolver->addAllowedTypes('ff', ['null', 'int']);
```
Comment (veewee) [inline src/GrumPHP/Task/Kahlan.php]: This ff option is duplicated.

File: doc/tasks/kahlan.md [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+          'no-header': ~
+          include: ~
+          exclude: ~
+          persistent: ~
+          cc: ~
+          autoclear: ~
+```
+
+Every options available are documented on [kahlan](https://kahlan.github.io/docs/cli-options.html)
```
Comment (veewee) [inline doc/tasks/kahlan.md]: Can you add more information about the options in this document? 
In the other tasks, we describe every possible option.
We would like to keep the documentation in the same format as the other tasks.

File: src/GrumPHP/Task/Kahlan.php (hunk 2 above)
Comment (Grafikart) [inline src/GrumPHP/Task/Kahlan.php]: Is there a way to add option automatically or do I have to use "addOptionalArgument" for every option ?

Comment (veewee) [inline src/GrumPHP/Task/Kahlan.php]: Currently you'll have to use `addOptionalArgument` for every option.
However, you could create an additional method for this on the arguments collection.

File: doc/tasks/kahlan.md (hunk 4 above)
Comment (Grafikart) [inline doc/tasks/kahlan.md]: Fixed

**Comment by Grafikart** [comment]
Fixed everything ,)

File: doc/tasks/kahlan.md (hunk 4 above)
Comment (Grafikart) [inline doc/tasks/kahlan.md]: I don't know why this review stay visible I fixed the problem :(

**Comment by veewee** [comment]
That looks great! Thanks for your work!

**Pull request merged**